crates/iceberg-catalog/src/service/storage/s3.rs

```diff
@@ -531,17 +534,17 @@ impl S3Profile {
                 creds.insert(&s3::SecretAccessKey(secret_access_key));
                 creds.insert(&s3::SessionToken(session_token));
             } else {
-                insert_pyiceberg_hack(&mut config);
+                push_fsspec_fileio_with_s3v4restsigner(&mut config);
                 remote_signing = true;
             }
         }
 
         if remote_signing {
             config.insert(&s3::RemoteSigningEnabled(true));
-            // Currently per-table signer uris are not supported by Spark.
-            // The URI is cached for one table, and then re-used for another.
-            // let signer_uri = CONFIG.s3_signer_uri_for_table(warehouse_id, namespace_id, table_id);
-            // config.insert("s3.signer.uri".to_string(), signer_uri.to_string());
```

Comment:
Does this mean we're limited to certain spark versions or does the location fallback help with that? Wouldn't this issue happen with older spark versions?

Reply:
Spark was never the problem - it just didn't support seperate "s3.signer.endpoint", but uris worked. The previous comment is wrong.